packages/python/plotly/_plotly_utils/utils.py

```diff
@@ -2,6 +2,8 @@
 import json as _json
 import sys
 import re
+from functools import reduce
+from numpy import cumsum
```

Comment:
not sure we want to import `numpy` here ... it's not a hard dependency of `plotly`

Reply:
Agreed, numpy shouldn't be a hard dependency here. Here's an alternative approach you can use for cumsum if you need it (https://realpython.com/python-reduce-function/#comparing-reduce-and-accumulate).